src/ray/core_worker/transport/out_of_order_actor_scheduling_queue.cc

```diff
@@ -86,7 +87,7 @@ void OutOfOrderActorSchedulingQueue::Add(
   // The reason why we don't run multiple attempts of the same
   // task concurrently is that it's not safe to assume user's
   // code can handle concurrent execution of the same actor method.
-  RAY_CHECK(boost::this_thread::get_id() == main_thread_id_);
+  RAY_CHECK(std::this_thread::get_id() == main_thread_id_);
```

Comment:
It's encouraged to use `RAY_CHECK_EQ`, because you're able to check the value when comparison fails.

Reply:
what's debug printer?

Reply:
I wrote a util class, which could print ANY type in C++, just like python: https://github.com/ray-project/ray/pull/51151

Reply:
Feel free to leave any comments